### PR TITLE
RDI-2036  Update documentation with some uses cases

### PR DESCRIPTION
**Pull request by rgonzalezfluendo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 This repository defines the process and tools to work on a project that is kept
 up-to-date with a remote (so-called upstream) repository.
 
+When working with open-source projects, new features can be needed
+before reaching upstream. Using plugins or hooks to add features is
```
Comment (turran) [inline README.md]: I don't understand this sentence:

> new features can be needed before reaching upstream

I'm assuming the features will reach upstream, but then:

> Using plugins or hooks to add features is generally recommended

This sentence is kind of contrary to the previous one. You want to go upstream but better do it through hooks?

File: README.md [hunk 2]
```diff
@@ -128,6 +135,16 @@ and is waiting for the community to be reviewed
...
 You cannot mix these groups.
 
+
+```mermaid
+stateDiagram-v2
+    pending --> merging
+    merging --> integrated
+```
+
+Iternal reviews done before doing the contibution must be done before the pending state to be sure
```
Comment (turran) [inline README.md]: I think this needs a rephrase. Do you mean that the internal reviews should be done before opening a contribution (PR/MR)? i.e in the `pending` state?

File: docs/workflows.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
+# Workflows for some common use cases 
+
+## 1.- Add a new feature at the end of the feature list.
```
Comment (turran) [inline docs/workflows.md]: Don't add custom numbering on markdown. Avoid any '1.-' and  just use the heading

File: docs/workflows.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+Manual actions to do by the dev:
```
Comment (turran) [inline docs/workflows.md]: Avoid 'dev', use 'Developer' instead

File: docs/workflows.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+* Create a new branch with the feature using the last feature as base.  
+* Create a new PR with the new branch to get the review.  
+
+Manual actions to do by the review:
```
Comment (turran) [inline docs/workflows.md]: Reviewer not review

File: docs/workflows.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+Automated actions done by GHA when the PR is approved.
+
+* Close PR with a comment.  
+* Execute the guw `update` command to sync the features and generate a new target.
+
+## 5.- Change or delete a commit of a pending feature because of an internal requirement without conflict.
+
+Similar to use case \#3 using a fixup commit to edit a previous commit of the feature or a revert commit to delete one. The goal is to make reviewing easier while ensuring a clean Git history after the review.  
+When the PR is approved the feature needs to be rebased by squashing fixup commits (autosquash)
```
Comment (turran) [inline docs/workflows.md]: Isn't this the `guw update`?

**Review by turran — CHANGES_REQUESTED** [review]

File: docs/workflows.md (hunk 5 above)
Comment (rgonzalezfluendo) [reply docs/workflows.md]: Done here https://github.com/fluendo/git-upstream-workflow/commit/ab28d87f116006f0ec7db41291a58a1cffce39f5

File: docs/workflows.md (hunk 4 above)
Comment (rgonzalezfluendo) [reply docs/workflows.md]: Done here https://github.com/fluendo/git-upstream-workflow/commit/366261db98104d62cd43397aab078396430af7df

File: docs/workflows.md (hunk 3 above)
Comment (rgonzalezfluendo) [reply docs/workflows.md]: I like numbering :disappointed: 

Done here https://github.com/fluendo/git-upstream-workflow/commit/5c697301c7748ff47fa837edee62234ae0c6c2f7

File: docs/workflows.md (hunk 6 above)
Comment (rgonzalezfluendo) [reply docs/workflows.md]: Done here https://github.com/fluendo/git-upstream-workflow/commit/fc8c1d30ad79930447415242227f48bf20d6cd61

File: README.md (hunk 2 above)
Comment (rgonzalezfluendo) [reply README.md]: To goal is a simple sentence to introduce the use cases in `docs/workflows.md`.


Done here: https://github.com/fluendo/git-upstream-workflow/commit/39d279d52f039a5d41f3d4aef4bb9080a551c369

File: README.md (hunk 1 above)
Comment (rgonzalezfluendo) [reply README.md]: Fixed https://github.com/fluendo/git-upstream-workflow/commit/18ddb694c43ad20698a420106003352a57b9aa5a

If you consider this paragraph to be too basic, let me know and proceed to delete it.

**Review by turran — APPROVED** [review]

**Pull request merged**